dll/win32/browseui/internettoolbar.cpp

```diff
@@ -714,45 +709,41 @@ HRESULT CInternetToolbar::CreateMenuBar(IShellMenu **pMenuBar)
 }
 
 HRESULT CInternetToolbar::LockUnlockToolbars(bool locked)
+{
+    if (locked != pSettings->fLocked)
+    {
+        pSettings->fLocked = locked;
+        pSettings->Save();
+        RefreshLockedToolbarState();
+        // TODO: refresh view menu?
```

Comment:
This should happen automatically if:
- the `pSettings` pointer points to the cached settings member of the window to which the toolbar belongs;
- the window' view menu "lock-toolbar" item state is set according to what is stored in the window's cached settings member.

Reply:
What are you trying to say? Every call to lock or unlock the toolbar goes through this function. At no other point in `browseui` is `fLocked` ever set to anything (except during initialization). `RefreshToolbarState()` only refreshes the toolbar window view after setting the locked state, it doesn't do any locking on its own.

Reply:
yeah this is what should happen (if pSettings is set to the correct thing as I explained).

Reply:
Right. I left that comment there because it was there before. No big deal.